### PR TITLE
feat: add group leadership list

### PR DESCRIPTION
**Pull request by rpcross**

### CODE REVIEW

File: ietf/group/views.py [hunk 1]
```diff
@@ -437,6 +438,39 @@ def prepare_group_documents(request, group, clist):
...
     return docs, meta, docs_related, meta_related
 
+def get_leadership(group_type):
+    people = Person.objects.filter(
+        role__name__slug='chair',
+        role__group__type=group_type,
+        role__group__state__slug__in=('active', 'bof', 'proposed')).distinct()
+    leaders = []
+    for person in people:
+        parts = person.name_parts()
```
Comment (jennifer-richards) [inline ietf/group/views.py:448]: Is this needed vs something like `person.plain_name()`? It'd be nice not to add to the variety of ways we handle names in various places. I guess this is to produce a "last name-first" list so maybe there's not much choice if that's needed for the Secretariat's purposes.

File: ietf/templates/group/group_leadership.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{% extends "base.html" %}
+{# Copyright The IETF Trust 2015, All Rights Reserved #}
```
Comment (jennifer-richards) [inline ietf/templates/group/group_leadership.html:2]: Copyright 2024

File: ietf/templates/group/group_leadership.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            {% for leader in leaders %}
+                <tr>
+                    <td>{{ leader.name }}</td>
+                    <td>{{ leader.groups }}</td>
+                </tr>
+            {% endfor %}
+        </tbody>
+    </table>
+{% endblock %}
+{% block js %}
```
Comment (jennifer-richards) [inline ietf/templates/group/group_leadership.html:35]: Can you omit the empty block?

**Review by jennifer-richards — CHANGES_REQUESTED** [review]
The tests should probably check that `rg` groups are handled correctly. As it stands, I think `rg` groups could be ignored and the tests would still pass. (`RoleFactory(group__type_id="rg", name_id="chair")` might be a useful tool for that.

It'd also be nice to test that active/bof/proposed groups are included but, e.g., conclude groups are ignored. Here, `RoleFactory(..., group__state_id="conclude", ...)` might help.

FYI, the existing group you're seeing ("Sops") is set up in `ietf/utils/test_data.py`.

Suggest running the Black styler over the new code. (Or at least using double-quotes instead of single)

File: ietf/group/views.py (hunk 1 above)
Comment (rpcross) [reply ietf/group/views.py:448]: Yes. The list is sorted by last name to match the way the meeting badges are sorted.

File: ietf/templates/group/group_leadership.html (hunk 3 above)
Comment (rpcross) [reply ietf/templates/group/group_leadership.html]: Done

File: ietf/templates/group/group_leadership.html (hunk 2 above)
Comment (rpcross) [reply ietf/templates/group/group_leadership.html]: Done

File: ietf/group/views.py [hunk 4]
```diff
@@ -437,6 +438,48 @@ def prepare_group_documents(request, group, clist):
 
     return docs, meta, docs_related, meta_related
 
+
+def get_leadership(group_type):
+    people = Person.objects.filter(
+        role__name__slug="chair",
```
Comment (rjsparks) [inline ietf/group/views.py:444]: Not asking for a change now, but do think about the future when we might want to add IAB programs to these pages - we'll need to look for other role names at that point.

**Review by jennifer-richards — APPROVED** [review]
Thanks for the changes!

**Pull request merged**